From 28a1433cc6342ee48cfea60b755c0e746b9ff548 Mon Sep 17 00:00:00 2001
From: Zhipeng Xie <xiezhipeng1@huawei.com>
Date: Fri, 2 Nov 2018 17:24:58 +0000
Subject: [PATCH 06/21] create-diff-object: don't create dynamic reloc for
 symbol exported by patch itself

when a patch export a new function, ___kcrctab+xxx has a reloc,
use origin reloc to get right symbol version.

Signed-off-by: Zhipeng Xie <xiezhipeng1@huawei.com>
---
 kpatch-build/create-diff-object.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/kpatch-build/create-diff-object.c b/kpatch-build/create-diff-object.c
index 78f5a7e..2cda233 100644
--- a/kpatch-build/create-diff-object.c
+++ b/kpatch-build/create-diff-object.c
@@ -3053,6 +3053,9 @@ static void kpatch_create_intermediate_sections(struct kpatch_elf *kelf,
 					if (!strcmp(sym_objname, "vmlinux"))
 						continue;
 
+					if (!strcmp(sym_objname, objname))
+						continue;
+
 					external = 1;
 				}
 			}
-- 
2.18.1
